fix: Prevent Manually Added Content in CLI Docs to be Overwritten

aboobakersiddiqr63 opened the pull request:
**Notes for Reviewers**

This PR fixes #9104 



**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ✓] Yes, I signed my commits.
 

<!--
Thank you for contributing to Meshery! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

l5io commented:
🚀 Preview for commit 73d569d231cb0408b90d13be2e8acaddacea3e29 at: https://652e0e5a9bb75234d233d311--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 3977467d338735dce77423bfeb1c69ead6ed277c at: https://652e1350165c1d3b995e5577--meshery-docs-preview.netlify.app

saurabh100ni commented:
Thanks for raising the PR, let's Discuss this in the Meshery Development Call today,
Please add this as an agenda in the Doc : [Doc](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit)

l5io commented:
🚀 Preview for commit d88c6dc8228a367830a41b148f7b5725e61c2abe at: https://652fbe74e46c95112af055f2--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit b43c313cb332c381ead1ea8d7a4215bc5cb41ce9 at: https://652fc5e21989fd1b7600dc93--meshery-docs-preview.netlify.app

alphaX86 reviewed (commented):
Great effort. Curious though. So to invoke this user would be applying them on the markdown page or in go program?

aboobakersiddiqr63 commented:
> Great effort. Curious though. So to invoke this user would be applying them on the markdown page or in go program?

Thanks @alphaX86. The user just needs to add the comment on the md file

l5io commented:
🚀 Preview for commit 997e5ebac94f1622ea07384895ee2411550a1e90 at: https://6533be2256d42a1ab87ea689--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 1322fda62d4f6f9b48cca583b92cbae173937579 at: https://6533d503e7d324284f546288--meshery-docs-preview.netlify.app

leecalcote commented on docs/pages/project/contributing/contributing-cli.md:
```diff
@@ -201,6 +201,10 @@ Though the command page is generated automatically by the Cobra CLI library, the
 - [cmds.yml](https://github.com/meshery/meshery/blob/master/docs/_data/mesheryctlcommands/cmds.yml) - The YAML file containing the data about the commands
 - [mesheryctl-commands.md](https://github.com/meshery/meshery/blob/master/docs/pages/reference/mesheryctl-commands.md) - The markdown page of the command reference documentation
 
+## Preserving Manually Added Documentation with Cobra CLI
+
+We use Cobra CLI and GitHub Actions to automate the generation of command documentation. To protect any manually added content and ensure it remains intact after regeneration, follow this format: {% raw %}{% include folder-name/file-name %}{% endraw %}. This format should reference an external file where your manual changes are stored. Any content added using this format will not be altered durin ...
+
```
```suggestion
## Preserving Manually Added Documentation

`mesheryctl` uses Cobra CLI and GitHub Actions to automate the generation of command documentation. On occasion, additional documentation beyond that included in the `mesheryctl` Golang files is ideal to capture and include in the CLI reference pages. Contributors are encouraged to add more usage examples, screenshots, video explainers and so forth to any of the CLI reference pages. To protect any manually added content and ensure it remains intact after regeneration, create a separate Jekyll `include` file. Follow file naming scheme outlined below:

```
{% raw %}{% include mesheryctl/<name-of-command> %}{% endraw %}
```

This format should reference an external file where your manual changes are stored. Any content added using this method will not be altered during the documentation generation process, but instead will be included post-auto doc generation. When making new changes or additions, understand that these additional details are positioned at the end their given CLI reference page, so bear this in mind as you organize and present your additional command details.

```

leecalcote commented on mesheryctl/doc/doc.go:
```diff
@@ -206,9 +215,10 @@ func GenMarkdownCustom(cmd *cobra.Command, w io.Writer) error {
 			})
 		}
-		buf.WriteString("Go back to [command reference index](/reference/mesheryctl/) ")
+		buf.WriteString("Go back to [command reference index](/reference/mesheryctl/), if you want to add content manually to the CLI documentation, please refer to the [instruction](/project/contributing/contributing-cli#preserving-manually-added-documentation-with-cobra-cli) for guidance.")
```
```suggestion
		buf.WriteString("Go back to [command reference index](/reference/mesheryctl/), if you want to add content manually to the CLI documentation, please refer to the [instruction](/project/contributing/contributing-cli#preserving-manually-added-documentation) for guidance.")
```

leecalcote reviewed (approved):
Looking good. 👍 

I think this new `include` method is a bit cleaner. Nice work.

saurabh100ni commented:
Let's discuss this on the Meshery dev call tomorrow at 7:30 pm IST / 9:00 am CT,
Please add this as your agenda item in Docs: https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit

leecalcote commented:
@aboobakersiddiqr63 will you send word when you have added an example to the Contributing portion of this PR?

aboobakersiddiqr63 commented:
> @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?

@leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned

l5io commented:
🚀 Preview for commit f34721d2363b2cfcd5ada6171b9c20d72b8ca630 at: https://653a162b65721959d62c954e--meshery-docs-preview.netlify.app

alphaX86 commented:
> > @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?
> 
> @leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned

I could not find an example actually. Could you share any markdown file utilizing this ability?

aboobakersiddiqr63 commented:
@alphaX86 

> > > @aboobakersiddiqr63 will you send word when you have added an include file example to the Contributing portion of this PR?
> > 
> > 
> > @leecalcote added an example of how to add manual content to the cli docs and updated other changes you have mentioned
> 
> I could not find an example actually. Could you share any markdown file utilizing this ability?

@alphaX86 I have added the example to contributing-cli.md file on how to proceed 
![image](https://github.com/meshery/meshery/assets/144536269/43929433-8904-490f-a576-e467abb8638a)

alphaX86 commented:
@aboobakersiddiqr63 noted. I'll try this and get back to you then 👍

leecalcote commented:
Gents, do we have an example for the docs?

Good to merge after this, yes?

abhijeetgauravm commented:
Hey @aboobakersiddiqr63  Let's discuss this on Meshery Development Meeting tomorrow at 8:30 PM IST / 9 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#).

MUzairS15 commented:
@alphaX86 Can you finish this off. // @aboobakersiddiqr63

alphaX86 commented:
> @alphaX86 Can you finish this off. // @aboobakersiddiqr63

I'll wrap this up @MUzairS15

l5io commented:
🚀 Preview for commit 979747d6e0d56dfb85983e5fe5585e34811fa09a at: https://655a10c9040c8637e4340e96--meshery-docs-preview.netlify.app

leecalcote commented:
> > @alphaX86 Can you finish this off. // @aboobakersiddiqr63
> 
> I'll wrap this up @MUzairS15

@alphaX86 good to go?

alphaX86 commented:
> > > @alphaX86 Can you finish this off. // @aboobakersiddiqr63
> > 
> > I'll wrap this up @MUzairS15
> 
> @alphaX86 good to go?

I'm testing the changes @leecalcote. Once they are good, we're good to merge this by EOD.

l5io commented:
🚀 Preview for commit bfb86114e8395cabfe67ea840db616844634c0d7 at: https://655e27ef2a036247e6b1d648--meshery-docs-preview.netlify.app

alphaX86 reviewed (dismissed):
Tested the changes 🟢. Made changes in doc page as well. LGTM 👍

alphaX86 commented:
Hold... one bug found in docs. Fixing it quick...

l5io commented:
🚀 Preview for commit 32a9aa9f4fcdd73e7122347e779aacc274a30096 at: https://655e333de2e9a44fe798016e--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit 13a9e0a9c36f64470ed678e80fddedf09dd64c4d at: https://655e34cf2a03625285b1d70e--meshery-docs-preview.netlify.app

l5io commented:
🚀 Preview for commit b623613241e962d56dfa6524510ae9f138d3f259 at: https://655e37c4b2a3c44dee74d4fe--meshery-docs-preview.netlify.app
